transports/httpclient/src/main/java/org/commonjava/maven/galley/transport/htcli/internal/HttpDownload.java

```diff
@@ -137,7 +142,25 @@ private void writeTarget()
             try
             {
                 final HttpEntity entity = response.getEntity();
-
+                /*
+                * If the server responded with 200 but the content length doesn't match,
+                * we will delete the file and return a 502 Gateway Error status to indicate
+                * the upstream server is having problems.
+                */
+                Header[] headers = response.getHeaders( "Content-Length" );
+                Collection<Integer> successStatuses = Collections.singleton( HttpStatus.SC_OK );
+                if ( headers[0] !=null && !headers[0].getValue().equals( "" ))
+                {
+                    long contentLength = Long.parseLong( headers[0].getValue().toString() );
+                    long targetLength = target.length();
+                    if ( contentLength != targetLength && successStatuses.contains( response.getStatusLine().getStatusCode() ) )
```

Comment:
I *think* we only call `writeTarget()` when we have a success status...in other words, that we've already checked for 200 status before we get here. If that's true, then we shouldn't need to check it again, right?

Reply:
Or if we want to check, why not directly HttpStatus.SC_OK==response.getStatusLine().getStatusCode()?

Reply:
@jdcasey yes you are right. we was checked the status in https://github.com/Commonjava/galley/blob/8eb579da6e06cb16e9abce0638437b5ea8781df1/transports/httpclient/src/main/java/org/commonjava/maven/galley/transport/htcli/internal/AbstractHttpJob.java#L112
@ligangty  Now HttpStatus.SC_OK just 200, maybe in future we will add other success status

Reply:
After a further check of the code, I've some concerns which I don't know if it is right. As this download job is executed in TransferManager.retrieve(), which will check if the the target transfer exists, so that means in this job the target transfer should not exist, and if this is true, why should we check the target transfer length with response header "content-length"?